### PR TITLE
RWA-2213: Add filter signature

### PR DESCRIPTION
**Pull request by rasikasri**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+                    "8d6cc5cf-c973-11eb-aaaa-000000000001", "1623278362400001"
+                ).toArray()
+            );
+    }
+
+    @Test
+    void should_return_task_list_and_count_when_filter_task_by_jurisdiction() {
+        List<RoleAssignment> roleAssignments = roleAssignmentsWithGrantTypeStandard();
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        indexRecord("8d6cc5cf-c973-11eb-aaaa-000000000001", "8d6cc5cf-c973-11eb-aaaa-100000000001");
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java:198]: same id's indexRecord?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+                    "8d6cc5cf-c973-11eb-aaaa-000000000001", "1623278362400001"
+                ).toArray()
+            );
+    }
+
+    @Test
+    void should_return_task_list_and_count_when_filter_task_by_role_category() {
+        List<RoleAssignment> roleAssignments = roleAssignmentsWithGrantTypeStandard();
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        indexRecord("8d6cc5cf-c973-11eb-aaaa-000000000001", "8d6cc5cf-c973-11eb-aaaa-200000000001");
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java:227]: same id's indexRecord?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+                    "8d6cc5cf-c973-11eb-aaaa-000000000001", "1623278362400001"
+                ).toArray()
+            );
+    }
+
+    @Test
+    void should_return_task_list_and_count_when_filter_task_by_location() {
+        List<RoleAssignment> roleAssignments = roleAssignmentsWithGrantTypeStandard();
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        indexRecord("8d6cc5cf-c973-11eb-aaaa-000000000001", "8d6cc5cf-c973-11eb-aaaa-300000000001");
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java:256]: same id's indexRecord?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+                    "8d6cc5cf-c973-11eb-aaaa-000000000001", "1623278362400001"
+                ).toArray()
+            );
+    }
+
+    @Test
+    void should_return_task_list_and_count_when_filter_task_by_work_type() {
+        List<RoleAssignment> roleAssignments = roleAssignmentsWithGrantTypeStandard();
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        indexRecord("8d6cc5cf-c973-11eb-aaaa-000000000001", "8d6cc5cf-c973-11eb-aaaa-400000000001");
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java:286]: same id's indexRecord?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+            );
+    }
+
+    @Test
+    void should_return_task_list_and_count_when_filter_task_by_state() {
+        List<RoleAssignment> roleAssignments = roleAssignmentsWithGrantTypeStandard();
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        indexRecord("8d6cc5cf-c973-11eb-aaaa-000000000001", "8d6cc5cf-c973-11eb-aaaa-000000000002");
+
+        SearchRequest searchRequest = SearchRequest.builder()
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java:171]: please review filter signature as some this test and some others use the exact same filter signature. is that the expectation?

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java (hunk 1 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java:198]: This is now addressed in RWA-2215 and all the records are indexed there.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java (hunk 2 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java:227]: This is now addressed in RWA-2215 and all the records are indexed there.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java (hunk 3 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java:256]: This is now addressed in RWA-2215 and all the records are indexed there.

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java (hunk 4 above)
Comment (rasikasri) [reply src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java:286]: This is now addressed in RWA-2215 and all the records are indexed there.

**Pull request merged**